central/securedclustercertgen/certificates_test.go

```diff
@@ -182,3 +229,57 @@ func (s *localScannerSuite) TestServiceIssueLocalScannerCerts() {
 	}
 	testutils.MustUpdateFeature(s.T(), features.ScannerV4, scannerV4Enabled)
 }
+
+// TestServiceIssueSecuredClusterCerts checks the issuance of certificates for secured clusters.
+func (s *securedClusterCertgenSuite) TestServiceIssueSecuredClusterCerts() {
+	allServiceTypeNames := make([]string, 0, allSupportedServiceTypes.Cardinality())
+	for _, serviceType := range allSupportedServiceTypes.AsSlice() {
+		allServiceTypeNames = append(allServiceTypeNames, serviceType.String())
+	}
+	allServiceTypeNamesSet := set.NewFrozenSet(allServiceTypeNames...)
```

Comment:
The original one is a string slice, and it's making it a frozen set. Then this set is being cloned before each test case, because in each one we remove items from it and expect it to be empty at the end.

Reply:
> The original one is a string slice, and it's making it a frozen set.

What do you consider to be the original? AFAICT this is copying from `allSupportedServiceTypes` which seems to be a frozenset as well, no? At least we're calling `AsSlice` on it on line 236.

> Then this set is being cloned before each test case, because in each one we remove items from it and expect it to be empty at the end.

Cloning is fine, my question is only about the code on lines 235-239.

Reply:
Oh, you're right! To my defense, the code became iteratively like this.

Reply:
It was actually transforming a FrozenSet of storage.ServiceType into a FrozenSet of strings, but also that is not needed.

Reply:
Done